examples/example_load_gen.py

```diff
@@ -55,38 +65,26 @@ class BasicPythonLoadGenContext:
 class BasicPythonLoadGen:
     cache_name = "python-loadgen"
 
-    def __init__(
-        self,
-        request_timeout_ms: int,
-        cache_item_payload_bytes: int,
-        max_requests_per_second: int,
-        number_of_concurrent_requests: int,
-        show_stats_interval_seconds: int,
-        total_seconds_to_run: int,
-    ):
+    def __init__(self, options: BasicPythonLoadGenOptions):
         self.logger = logging.getLogger("load-gen")
         self.auth_token = os.getenv("MOMENTO_AUTH_TOKEN")
         if not self.auth_token:
             raise ValueError("Missing required environment variable MOMENTO_AUTH_TOKEN")
-        self.request_timeout_ms = request_timeout_ms
-        self.max_requests_per_second = max_requests_per_second
-        self.number_of_concurrent_requests = number_of_concurrent_requests
-        self.show_stats_interval_seconds = show_stats_interval_seconds
-        self.total_seconds_to_run = total_seconds_to_run
-        self.cache_value = "x" * cache_item_payload_bytes
-        self.request_interval_ms = number_of_concurrent_requests / max_requests_per_second * 1000
+        self.options = options
+        self.cache_value = "x" * options.cache_item_payload_bytes
+        self.request_interval_ms = options.number_of_concurrent_requests / options.max_requests_per_second * 1000
 
     async def run(self) -> None:
         cache_item_ttl_seconds = 60
         async with scc.SimpleCacheClient(
-            self.auth_token, cache_item_ttl_seconds, self.request_timeout_ms
+            self.auth_token, cache_item_ttl_seconds, self.options.request_timeout_ms
         ) as cache_client:
             try:
                 await cache_client.create_cache(BasicPythonLoadGen.cache_name)
             except momento.errors.AlreadyExistsError:
                 self.logger.info(f"Cache with name: {BasicPythonLoadGen.cache_name} already exists.")
 
-            load_gen_context = BasicPythonLoadGenContext(
+            self.context = BasicPythonLoadGenContext(
```

Comment:
i'm a little more iffy about this.  i really don't like having member variables that contain mutable state in async/concurrent contexts because it can make the coordination semantics more difficult (and slow).

In python this is a non-issue because of the GIL so I'm okay with this if you prefer it.  But in the languages that support parallelism we will probably want to scrutinize this sort of change more thoroughly.  Making the functions as pure as possible (not relying on side effects / object state, and preferring arguments) can really make a huge difference in code maintainability.

Reply:
Yes, there's no threading here to worry about. And if there was we'd need to make sure they're all using the same BasicPythonLoadGenContext regardless.

If there was, I'd start by extracting several classes out of BasicPythonLoadGen; it does too much.

Reply:
> If there was, I'd start by extracting several classes out of BasicPythonLoadGen; it does too much.

Yes, if it was production code I would not have designed it this way.  The intent is for users to be able to easily play with it and modify it if they like, to test different operations or payload types.  That is why it is one class in one file.

Reply:
> Yes, there's no threading here to worry about. And if there was we'd need to make sure they're all using the same BasicPythonLoadGenContext regardless.

Not necessarily.  I would probably have each worker keep track of its own state and then have the main thread aggregate them so that there was no lock contention or other issues related to shared mutable state to deal with.  So I would very likely be trying hard to avoid storing the mutable state in a widely accessible class variable.

But this is partially a stylistic consideration, and again I acknowledge that it is not relevant for this particular Python environment so I am okay with the change if you prefer it.  I'm just trying to give you a heads up on my thoughts as we move into .NET, Java, go, Rust, etc.